eZ/Publish/Core/FieldType/Image/IO/Legacy.php

```diff
@@ -149,7 +150,13 @@ public function loadBinaryFile($binaryFileId)
     public function loadBinaryFileByUri($binaryFileUri)
     {
         try {
-            return $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+            $image = $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+
+            if ($image instanceof MissingBinaryFile) {
+                throw new InvalidArgumentException('binaryFileUri', sprintf("Can't find file with url {0}", $binaryFileUri));
```

Comment:
Not wrong, though I think mostly we don't use sprintf unless the formatting functionality is needed.
https://github.com/ezsystems/ezpublish-kernel/blob/7a0b16e548d673b3227a90195b603fc6e1ba3aac/eZ/Publish/Core/IO/IOService.php#L85